up-linux-streamer/examples/mE_client.rs

```diff
@@ -0,0 +1,108 @@
+const CLIENT_UE_VERSION_MAJOR: u8 = 1;
+const CLIENT_RESOURCE_ID: u16 = 0;
+
+const REQUEST_TTL: u32 = 1000;
+
+struct ServiceResponseListener;
+
+#[async_trait]
+impl UListener for ServiceResponseListener {
+    async fn on_receive(&self, msg: UMessage) {
+        println!("ServiceResponseListener: Received a message: {msg:?}");
+
+        let Some(payload_bytes) = msg.payload else {
+            panic!("No payload bytes");
+        };
+
+        let Ok(hello_response) = HelloResponse::parse_from_bytes(&payload_bytes) else {
+            panic!("Unable to parse into HelloResponse");
+        };
+
+        println!("Here we received response: {hello_response:?}");
+    }
+
+    async fn on_error(&self, err: UStatus) {
+        println!("ServiceResponseListener: Encountered an error: {err:?}");
+    }
+}
+
+#[tokio::main]
+async fn main() -> Result<(), UStatus> {
+    env_logger::init();
+
+    println!("mE_client");
+
+    let crate_dir = env!("CARGO_MANIFEST_DIR");
+    // TODO: Make configurable to pass the path to the vsomeip config as a command line argument
+    let vsomeip_config = PathBuf::from(crate_dir).join("vsomeip-configs/mE_client.json");
+    let vsomeip_config = canonicalize(vsomeip_config).ok();
+    trace!("vsomeip_config: {vsomeip_config:?}");
+
+    // There will be a single vsomeip_transport, as there is a connection into device and a streamer
+    // TODO: Add error handling if we fail to create a UPTransportVsomeip
+    let client: Arc<dyn UTransport> = Arc::new(
+        UPTransportVsomeip::new_with_config(
+            &CLIENT_AUTHORITY.to_string(),
+            CLIENT_UE_ID,
+            &vsomeip_config.unwrap(),
+        )
+        .unwrap(),
+    );
+
+    let source = UUri {
+        authority_name: CLIENT_AUTHORITY.to_string(),
+        ue_id: CLIENT_UE_ID as u32,
+        ue_version_major: CLIENT_UE_VERSION_MAJOR as u32,
+        resource_id: CLIENT_RESOURCE_ID as u32,
+        ..Default::default()
+    };
+    let sink = UUri {
+        authority_name: SERVICE_AUTHORITY.to_string(),
+        ue_id: SERVICE_UE_ID as u32,
+        ue_version_major: SERVICE_UE_VERSION_MAJOR as u32,
+        resource_id: SERVICE_RESOURCE_ID as u32,
+        ..Default::default()
+    };
+
+    let service_response_listener: Arc<dyn UListener> = Arc::new(ServiceResponseListener);
+    client
+        .register_listener(&sink, Some(&source), service_response_listener)
+        .await?;
+
+    let mut i = 0;
+    loop {
+        tokio::time::sleep(Duration::from_millis(1000)).await;
+
+        let hello_request = HelloRequest {
+            name: format!("me_client@i={}", i).to_string(),
+            ..Default::default()
+        };
+        i += 1;
+
+        let request_msg = UMessageBuilder::request(sink.clone(), source.clone(), REQUEST_TTL)
+            .build_with_protobuf_payload(&hello_request)
+            .unwrap();
+        println!("Sending Request message:\n{request_msg:?}");
+
+        client.send(request_msg).await?;
```

Comment:
use `up_rust::communication::InMemoryRpcClient`

Reply:
I'll bring this in as a part of this PR: https://github.com/eclipse-uprotocol/up-streamer-rust/pull/26

Will mark this as resolved and to be tackled over there